api/v1alpha1/tls_types.go

```diff
@@ -15,6 +15,20 @@ type ClientTLSSettings struct {
 	// +optional
 	ClientValidation *ClientValidationContext `json:"clientValidation,omitempty"`
 	TLSSettings      `json:",inline"`
+
+	// SessionTimeout determines the maximum lifetime of a TLS session.
+	// https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#SSL_DEFAULT_SESSION_TIMEOUT
+	// Default: 7200s
+	// +optional
+	SessionTimeout *gwapiv1.Duration `json:"sessionTimeout,omitempty"`
+
+	// SessionResumptionSettings determine the proxy's supported TLS session resumption option.
+	// By default, Envoy Gateway does not enable session resumption. Use sessionResumption to
+	// enable stateful and stateless session resumption. Users should consider security impacts
+	// of different resumption methods. Performance gains from resumption are diminished when
+	// Envoy proxy is deployed with more than one replica.
+	// +optional
+	SessionResumptionSettings *SessionResumption `json:"sessionResumption,omitempty"`
```

Comment:
can we use following structure?
```
session:
  timeout:
  resumption:
```

if there're more than 1 property with session prefix, prefer to move them into a struct.

Reply:
Should the `timeout` be part of the SessionResumption setting? 

* https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/tls.proto#envoy-v3-api-msg-extensions-transport-sockets-tls-v3-downstreamtlscontext
* https://datatracker.ietf.org/doc/html/rfc5077#section-5.6

Reply:
Well, it seems that different implementations have different meanings for session timeout. 

[BoringSSL](https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#SSL_DEFAULT_SESSION_TIMEOUT) seems to treat this as a hard timeout after which a new handshake is required:
>  This is how long we are willing to use the secret to encrypt traffic without fresh key material.

In envoy, this is also used as a hint on session tickets. 

[OpenSSL](https://docs.openssl.org/master/man3/SSL_get_default_timeout) seems to treat this as a soft timeout after which resumption cannot occur, but an active session may still continue (?):
> Whenever a new session is negotiated, it is assigned a timeout value, after which it will not be accepted for session reuse. 

@arkodg requested that we leave `sessionTimeout` out of this PR. I will change the API to @zirain's proposal, to support for future addition of session-related settings that are not specific to resumption.